Switch to new JSON-RPC dependency

chrisrico opened the pull request:
[This one](https://github.com/Polve/JavaBitcoindRpcClient) is based on the same project, but is more actively maintained.

Also fixed some dependency issues that only appeared when using the public repositories (not the GB ones).

This change makes BATMBitcoindRPCWallet bitcoind 0.17+ friendly. If no accountName is set in the config, it will call getBalance(), sendToAddress(), and getNewAddress() instead of getBalance(account), sendFrom(), and getAddressesByAccount().

rpanak-generalbytes commented:
@chrisrico Sorry for the mess with the dependencies. Anyway, it should be fixed now.

Please do the following changes regarding dependencies and checksums:
- bitrafael 1.0.13 should be available, so revert your downgrade (i.e. use 1.0.13)
- there should be no need for adding or changing checksums for existing dependencies (bitrafael, dashj, litecoinj). They should work right away if you update from master branch
- instead of using JavaBitcoindRpcClient's jar, use the proper Maven dependency from JCenter. Basically this should mean you have to remove the JAR from your commit and change the 1.0.0-SNAPSHOT version to a properly released version (see https://bintray.com/bintray/jcenter/wf.bitcoin%3AJavaBitcoindRpcClient, you'll probably want to use the newest one). You'll of couse have to update the checksums accordingly (remove 1.0.0-SNAPSHOT and add the one you start using).

The final version of your diff regarding checksums and dependencies should only switch the Bitcoin RPC client dependency and the corresponding checksums.

chrisrico commented:
> Please do the following changes regarding dependencies and checksums

Ok, I've removed the commit changing the other checksums.

> instead of using JavaBitcoindRpcClient's jar, use the proper Maven dependency from JCenter

I would, but it's very out of date (last updated July). As an alternative, I've added a reference to the JitPack repository. They will resolve a Github repo and build the artifact if it doesn't exist yet. Let me know what you think, it seems to work well. I have it based on a current commit, but it could be based on a release tag or branch.

generalbytes commented:
Hi Chris,
Thank you for contributing this improvement to our product. I really appreciate it.
